# Thomas Hoffmann, EBML Heidelberg, structures-it@embl.de, 2024/06
# allow to inject cuda compute capabilities and cfalgs
diff -ru AreTomo3-c6c5a13f00e6d94327bb188dd683f596433d3c17/makefile11 AreTomo3-c6c5a13f00e6d94327bb188dd683f596433d3c17_makefile/makefile11
--- AreTomo3-c6c5a13f00e6d94327bb188dd683f596433d3c17/makefile11	2024-06-20 08:29:57.000000000 +0200
+++ AreTomo3-c6c5a13f00e6d94327bb188dd683f596433d3c17_makefile/makefile11	2024-06-25 11:14:57.668964213 +0200
@@ -246,18 +246,23 @@
 	$(CUCPPS)
 OBJS = $(patsubst %.cpp, %.o, $(SRCS))
 #-------------------------------------
-CC = g++ -std=c++11
-CFLAG = -c -g -pthread -m64
+#CC = g++ -std=c++11
+CFLAG = -c -g -pthread -m64 $(CFLAGS)
 NVCC = $(CUDAHOME)/bin/nvcc -std=c++11
-CUFLAG = -Xptxas -dlcm=ca -O2 \
-	-gencode arch=compute_86,code=sm_86 \
-	-gencode arch=compute_80,code=sm_80 \
-	-gencode arch=compute_75,code=sm_75 \
-	-gencode arch=compute_70,code=sm_70 \
-	-gencode arch=compute_52,code=sm_52 \
-        -gencode arch=compute_53,code=sm_53 \
-        -gencode arch=compute_60,code=sm_60 \
-        -gencode arch=compute_61,code=sm_61 
+SPACE= ' '
+SEMI= ;
+ENCODES = $(foreach x,$(subst $(SEMI),$(SPACE),$(CUDACC)),-gencode arch=compute_$x,code=sm_$x)
+CUFLAG = -Xptxas -dlcm=ca -O2 $(GENCODES)
+
+CUFLAG = -Xptxas -dlcm=ca -O2 $(GENCODES)
+#	-gencode arch=compute_86,code=sm_86 \
+#	-gencode arch=compute_80,code=sm_80 \
+#	-gencode arch=compute_75,code=sm_75 \
+#	-gencode arch=compute_70,code=sm_70 \
+#	-gencode arch=compute_52,code=sm_52 \
+#        -gencode arch=compute_53,code=sm_53 \
+#        -gencode arch=compute_60,code=sm_60 \
+#        -gencode arch=compute_61,code=sm_61 
 #------------------------------------------
 cuda: $(CUCPPS)
 
@@ -272,6 +277,7 @@
 	-lcufft -lcudart -lcuda -lnvToolsExt -ltiff -lc -lm -lpthread \
 	-o AreTomo3
 	@echo AreTomo3 has been generated.
+	@echo used cuda gencodes: $(GENCODES)
 
 %.cpp: %.cu
 	@echo "-----------------------------------------------"
